### PR TITLE
Config changes to switch EDS to gRPC

### PR DESCRIPTION
**Pull request by gnirodi**
Switches EDS to use gRPC

### CODE REVIEW

File: tools/deb/envoy_bootstrap_tmpl.json [hunk 1]
```diff
@@ -33,10 +38,10 @@
     "deprecated_v1": {
       "sds_config": {
         "api_config_source": {
-          "api_type": "REST_LEGACY",
+          "api_type": "GRPC",
           "refresh_delay": {{ .refresh_delay }},
           "cluster_names": [
-            "rds"
+            "eds-grpc"
```
Comment (kyessenov) [inline tools/deb/envoy_bootstrap_tmpl.json]: change it to xds-grpc, not specific to eds.

Comment (costinm) [reply tools/deb/envoy_bootstrap_tmpl.json]: +1 - but can be done in the next PR.

**Comment by kyessenov** [comment]
It's a simple change. There is no reason to call eds-grpc. Would you agree
with that?

File: tests/istio.mk [hunk 2]
```diff
@@ -86,7 +86,7 @@ test/minikube/auth/e2e_pilot: istioctl generate_yaml
 	kubectl create ns istio-test || true
 	go test -test.v -timeout 20m ./tests/e2e/tests/pilot -args \
 		-hub ${HUB} -tag ${TAG} \
-		--skip-cleanup --mixer=true --auth=enable \
```
Comment (ldemailly) [inline tests/istio.mk]: ahem, how about no ?

Comment (rshriram) [reply tests/istio.mk]: please don't remove this line. We would spend 2x testing time - testing with and without auth. Also, why do we have to change test makefile as part of this PR?

File: install/kubernetes/templates/istio-ingress.yaml.tmpl [hunk 3]
```diff
@@ -81,6 +84,8 @@ spec:
             fieldRef:
               apiVersion: v1
               fieldPath: metadata.namespace
+        - name: PILOT_GRPC_PORT
+          value: "15443"
```
Comment (costinm) [inline install/kubernetes/templates/istio-ingress.yaml.tmpl]: Default value out of range - we trying to use 150xx addresses in a more compact range.
How about 15010 ?

**Pull request closed without merging**